File: Permutations/Permutations.java
public class Solution {

    public List<List<Integer>> permute(int[] num) {
        List<List<Integer>> result = new ArrayList<List<Integer>>();

        result.add(new ArrayList<Integer>());
        List<List<Integer>> current = new ArrayList<List<Integer>>();
        for (int i = 0; i < num.length; ++i) {
          
            current.clear();
            for (List<Integer> l : result) {
           
                for (int j = 0; j < l.size()+1; ++j) {
                  
                    l.add(j, num[i]);
                    ArrayList<Integer> temp = new ArrayList<Integer>(l);
                    current.add(temp);
                    l.remove(j); 
                }
            }
        result = new ArrayList<List<Integer>>(current);
        }
        return result;
    }

}

File: Permutations/permutations.cpp
class Solution {
public:
    vector<vector<int>> permute(vector<int>& nums) {
        vector<vector<int>> ans;
        vector<int> temp=nums;
        ans.push_back(nums);
        sort(nums.begin(),nums.end());
        do{
            if(nums!=temp)
                ans.push_back(nums);
        }while(next_permutation(nums.begin(),nums.end()));
        return ans;
    }
};
